### PR TITLE
#169: linux/mac setup fails with *.xz files

### PR DESCRIPTION
**Pull request by sobkowiak**
The `functions` file has been extended by handling the `.xz` files with `tar xf file`

### CODE REVIEW

File: scripts/src/main/resources/scripts/functions [hunk 1]
```diff
@@ -351,6 +351,9 @@ function doExtract() {
   then
     doRunCommand "unzip -q '${1}' -d '${target_dir}'"
   elif [ "${ext}" = ".tar" ]
+  then
+    doRunCommand "tar xf '${1}' -C '${target_dir}'"
+  elif [ "${ext}" = ".xz" ]
   then
     doRunCommand "tar xf '${1}' -C '${target_dir}'"
```
Comment (hohwille) [inline scripts/src/main/resources/scripts/functions]: shouldnt it be "tar xJf ..." for *.xz files?
BTW: Which software tool is using this odd format?

Comment (hohwille) [reply scripts/src/main/resources/scripts/functions]: Will that be supported in git-bash (mingw) after all?

**Comment by hohwille** [comment]
@sobkowiak Thanks for your improvement. I am happy to receive PRs with features or cases I am not aware of. 👍 
So I am happy to include this change. However, could you comment on my little review question?

File: scripts/src/main/resources/scripts/functions (hunk 1 above)
Comment (sobkowiak) [reply scripts/src/main/resources/scripts/functions]: tar detects the file format, but you are right, maybe adding the option J would be better

Comment (sobkowiak) [reply scripts/src/main/resources/scripts/functions]: I"ll check the bash-git tomorrow

Comment (sobkowiak) [reply scripts/src/main/resources/scripts/functions]: > Will that be supported in git-bash (mingw) after all?

As you can see in the linked issue (#169) nodejs is provided as .[xz](https://en.wikipedia.org/wiki/Xz) file (`node.js-v10.16.0-linux.tar.xz`) I can see more and more often this file format.

Comment (sobkowiak) [reply scripts/src/main/resources/scripts/functions]: > 
> 
> I"ll check the bash-git tomorrow

Ye, it works under git-bash

**Comment by markusschuh** [comment]
More with respect to the original code of doExtract in functions - not specifically this change:

For both bsdtar (macOS) as well as gnu-tar it is recommended to let them autodectect archive formats of existing archives and to not specify the decompress format as additional option.
A simple 

 tar xf $archive

will work for all archives, first "tar"ed, then - optionally - compressed with either gzip, bzip2 or xz.

The current code within "functions" tries to redo the internal tar logic within the outside scripting. This looks ineffective. Also it unnecessarily introduces small glitches - e.g the *.tar.bz2 case is not yet handled properly. I would just check for an acceptable extension

`ext=$( echo $filename | sed -E -e 's/^.*(\.(zip|dmg|tar|tar\.gz|tgz|tar.bz2|tbz2|tar.xz))$/\1/' )`

and then differentiate between the cases zip, dmg, empty string or else - where the else case is always this

`doRunCommand "tar xf '${1}' -C '${target_dir}'"`

**Comment by hohwille** [comment]
@markusschuh thanks for your feedback. I agree that we should solve this more general and especially support `*.tar.*` this way. I assume that the autodetection of tar is (even more) reliable and works on all platforms. Then I would not merge this PR and recreate the change from scratch.

**Comment by hohwille** [comment]
Replaced with PR #176

**Pull request closed without merging**